core/src/main/scala/kafka/coordinator/transaction/TransactionCoordinator.scala

```diff
@@ -564,41 +627,44 @@ class TransactionCoordinator(txnConfig: TransactionConfig,
                 else
                   PrepareAbort
 
-                // Maybe allocate new producer ID if we are bumping epoch and epoch is exhausted
-                val nextProducerIdOrErrors =
-                  if (clientTransactionVersion.supportsEpochBump() && !txnMetadata.pendingState.contains(PrepareEpochFence) && txnMetadata.isProducerEpochExhausted) {
-                    try {
-                      Right(producerIdManager.generateProducerId())
-                    } catch {
-                      case e: Exception => Left(Errors.forException(e))
-                    }
+                generateTxnTransitMetadataForTxnCompletion(nextState, false)
+              case CompleteCommit =>
+                if (currentTxnMetadataIsAtLeastTransactionsV2) {
+                  if (txnMarkerResult == TransactionResult.COMMIT) {
+                    if (isRetry)
```

Comment:
nIt: we don't have to change this but can we ever get here if this is not a retry?

Reply:
Yes, it is possible. We have a UT covering this.

Reply:
I'm wondering if it could be useful to specify commit vs abort instead of just if commit. It gets a little confusing to understand the if cases here between v1,v2, retries, and commit/abort

Reply:
Added comments for each case.